internal/server/grants.go

```diff
@@ -25,6 +26,13 @@ func (r ListGrantsResponse) SetHeaders(h http.Header) {
 }
 
 func (a *API) ListGrants(c *gin.Context, r *api.ListGrantsRequest) (*ListGrantsResponse, error) {
+	rCtx := getRequestContext(c)
+
+	rCtx.AddLogFields = append(rCtx.AddLogFields, func(event *zerolog.Event) {
+		event.Int64("lastUpdateIndex", r.LastUpdateIndex)
+	})
```

Comment:
I add this separately from `numGrants` so that if the request errors and exits early we still get this context added to the log entry.